src/main/java/io/supertokens/webserver/api/session/SessionRemoveAPI.java

```diff
@@ -101,28 +110,41 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
         }
 
         if (userId != null) {
+            Storage storage = null;
             try {
-                TenantIdentifier tenantIdentifier = getTenantIdentifier(req);
-                Storage storage = getTenantStorage(req);
-
                 String[] sessionHandlesRevoked;
+
                 if (revokeAcrossAllTenants) {
+                    AppIdentifier appIdentifier = getAppIdentifier(req);
```

Comment:
here we need to find the storage that belongs to the input user without enforcing public tenant (cause int he backend sdk, we are always passing tenant id in the path, which we should not have, but to avoid changing that, we do this here)

Reply:
Copy the code of enforcePublicTenantAndGetStorageAndUserIdMappingForAppSpecificApi, but do not enforce public tenant

Reply:
Done